Assets/Code/Gameplay/TankObject.cs

```diff
@@ -0,0 +1,129 @@
+﻿using System.Collections;
+using System.Collections.Generic;
+using UnityEngine;
+
+public class TankObject : GameplayObject
+{
+
+    public EnemyGunScript m_EnemyGun;
+
+    private bool m_getsBonus;
+    private bool m_explosionDeath;
+
+    private const int k_enemyPoints = 1000;
+    private const int k_bonusPoints = 1000;
+    private const int k_explosionBonusPoints = 25;
+    private const float k_range = 20;
+
+    // Use this for initialization
+    void Start()
+    {
+        HandleBirth();
+        InvokeRepeating("HandleMovement", 5, 5);
+    }
+
+    // Update is called once per frame
+    void Update()
+    {
+    }
+
+    protected override void HandleBirth()
+    {
+        base.HandleBirth();
+        m_health = 20000;
+    }
+
+    protected override void HandleMovement()
+    {
+        Vector3 playerPosition = FindObjectOfType<PlayerObject>().gameObject.transform.position;
+
+        if (playerPosition == null)
+        {
+            Debug.LogError("Unable to find player in the scene");
+        }
+
+        transform.position = JumpWithinRange(transform.position, playerPosition);
+    }
+
+    private void OnCollisionEnter(Collision collision)
+    {
+        GameObject collisionObject = collision.gameObject;
+
+        switch (collisionObject.tag)
+        {
+            case "BulletTag":
+                BulletObject bullet = collisionObject.GetComponent<BulletObject>();
+
+                // This enemy is being hit by a bullet.  Reduce its health by the strength of the bullet
+                m_health -= bullet.m_BulletStrength;
+
+                if (m_health <= 0)
+                {
+                    // Player gets bonus for killing blow bouncing bullet off wall first
+                    m_getsBonus = bullet.m_hitWall;
+
+                    // This enemy has taken too much damage, kill it.
+                    HandleDeath();
+                }
+                break;
+            case "GroundTag":
+                break;
+            case "WallTag":
+                break;
+            default:
+                Debug.Log("Encountered an unhandled tag: " + collisionObject.tag);
+                break;
+        }
+    }
+
+    protected override void HandleDeath()
+    {
+        if (!m_isDying)
+        {
+            m_isDying = true;
+            Destroy(this.gameObject);
+
+            ScoreManager.AddScore(k_enemyPoints);
+            if (m_getsBonus)
+            {
+                ScoreManager.AddScore(k_bonusPoints);
+            }
+            if (m_explosionDeath)
+            {
+                ScoreManager.AddScore(k_explosionBonusPoints);
+            }
+        }
+    }
+
+    private void OnTriggerEnter(Collider other)
+    {
+        GameObject otherObject = other.gameObject;
```

Comment:
Eventually I want to move everything over to using tags since we really don't use rigid bodies on enemies too much.